src/pages/VeSDL/index.tsx

```diff
@@ -4,52 +4,245 @@ import {
+      )
+
+      const veSdlToken = await votingEscrowContract?.locked(account)
+      setLockedSDLVal(veSdlToken?.amount || Zero)
+    }
+  }, [account, sdlContract, votingEscrowContract])
+
+  function calculateLockAmount() {
+    if (!unlockDate) return
+    if (isNaN(unlockDate.valueOf())) return
+    const currentTimestamp = getUnixTime(new Date())
+    const expireTimestamp = getUnixTime(unlockDate)
+    const totalAmount = sdlTokenValue.add(lockedSDLVal)
+    const roundedExpireTimestamp = BigNumber.from(expireTimestamp)
+      .div(WEEK)
+      .mul(WEEK)
+
+    const estLockValue = totalAmount
+      .mul(BigNumber.from(roundedExpireTimestamp).sub(currentTimestamp))
+      .div(MAXTIME)
+    return formatUnits(estLockValue)
+  }
+
+  const getPenalty = () => {
+    if (!lockEnd) return
+    const leftTime = getUnixTime(lockEnd) - getUnixTime(new Date())
+    const penaltyRatio = Math.min(
+      (MULTIPLIER * 3) / 4,
+      (MULTIPLIER * leftTime) / MAXTIME,
+    ).toFixed()
+
+    const penaltyAmount = lockedSDLVal
+      .mul(BigNumber.from(penaltyRatio))
+      .div(BigNumber.from(MULTIPLIER))
+    return formatUnits(penaltyAmount)
+  }
+
+  const handleLock = async () => {
+    if (!account || !chainId || !votingEscrowContract?.address) return
+    try {
+      const unlockTimeStamp = unlockDate ? getUnixTime(unlockDate) : null
+      if (sdlTokenValue.gt(Zero))
+        await sdlContract?.approve(votingEscrowContract?.address, sdlTokenValue)
+
+      if (sdlTokenValue.gt(Zero) && unlockTimeStamp) {
+        if (lockedSDLVal.isZero()) {
+          const txn = await votingEscrowContract?.create_lock(
+            sdlTokenValue,
+            unlockTimeStamp,
+          )
+          await txn.wait()
+        } else {
+          const txnIncreaseAmount = await votingEscrowContract.increase_amount(
+            sdlTokenValue,
+          )
+          await txnIncreaseAmount.wait()
+          const txnIncreaseUnlockTime =
+            await votingEscrowContract.increase_unlock_time(
+              BigNumber.from(unlockTimeStamp),
+            )
+          await txnIncreaseUnlockTime.wait()
+        }
+
+        setUnlockDate(null)
+      } else if (
+        sdlTokenValue.gt(Zero) &&
+        !unlockTimeStamp &&
+        !veSdlTokenVal.isZero()
+      ) {
+        // Deposit additional SDL into and existing lock
+        const txn = await votingEscrowContract.increase_amount(sdlTokenValue)
+        await txn.wait()
+        setUnlockDate(null)
+      } else if (
+        !sdlTokenValue.gt(Zero) &&
+        unlockTimeStamp &&
+        !veSdlTokenVal.isZero()
+      ) {
+        // Extend the unlock time on a lock that already exists
+        const txn = await votingEscrowContract.increase_unlock_time(
+          BigNumber.from(unlockTimeStamp),
+        )
+        await txn.wait()
+        setUnlockDate(null)
+      }
+      void fetchData()
+    } catch (err) {
+      console.log(err)
+    }
+  }
+
+  const handleUnlock = async () => {
```

Comment:
prefer to use `useCallback`

Reply:
I don't think using useCallback is necessary for this function.
1. Button component is not component wrapped with useMemo, so it will re-rendered with the change of state of `VeSDL` component.
2. Also, handleUnlock is not passed as dependency in Button, so I think using useCallback is not necessary for this handler.